### DISCUSSION TITLE
Why does upsert create two queries?

### DISCUSSION THREAD

**Discussion by GasimGasimzada** [prisma/prisma#7325] [Q&A] [answered]
When running upsert, instead of using `INSERT ... ON CONFLICT ...`, `INSERT ... ON DUPLICATE KEY ...`, it is doing two queries -- SELECT + INSERT/UPDATE. Is there a specific reason for it? Can this be optimized; so, it sends one query in select data sources (e.g postgres, MySQL support them; mongo also has `{ upsert: true }` but I am not familiar with Mongo and its implementation in Prisma; so, can't say much about that).

**Comment by janpio** [comment, marked as answer]
If the end result is the correct one, this is currently not really in our focus. We have areas where Prisma is not doing the right thing, and those get prioritized over suboptimal query strategies. You can still open an issue of course with more information, and we will label it - but then it will probably sit there for quite some time.
